dep-graph/client-e2e/project.json

```diff
@@ -3,31 +3,38 @@
   "sourceRoot": "dep-graph/client-e2e/src",
   "projectType": "application",
   "targets": {
-    "e2e-disabled": {
+    "e2e-base": {
       "executor": "@nrwl/cypress:cypress",
       "options": {
-        "cypressConfig": "dep-graph/client-e2e/cypress.json",
         "tsConfig": "dep-graph/client-e2e/tsconfig.e2e.json",
-        "devServerTarget": "dep-graph-client:serve-for-e2e",
         "baseUrl": "http://localhost:4200"
-      }
-    },
-    "e2e-watch-disabled": {
-      "executor": "@nrwl/cypress:cypress",
-      "options": {
-        "cypressConfig": "dep-graph/client-e2e/cypress-watch-mode.json",
-        "tsConfig": "dep-graph/client-e2e/tsconfig.e2e.json",
-        "devServerTarget": "dep-graph-client:serve-for-e2e:watch",
-        "baseUrl": "http://localhost:4200"
-      }
+      },
+      "configurations": {
+        "dev": {
+          "cypressConfig": "dep-graph/client-e2e/cypress.json",
+          "devServerTarget": "dep-graph-client:serve"
+        },
+        "watch": {
+          "cypressConfig": "dep-graph/client-e2e/cypress-watch-mode.json",
+          "devServerTarget": "dep-graph-client:serve:watch"
+        },
+        "release": {
+          "cypressConfig": "dep-graph/client-e2e/cypress-release.json",
+          "devServerTarget": "dep-graph-client:serve:release"
+        }
+      },
+      "defaultConfiguration": "dev"
     },
-    "e2e-release-disabled": {
-      "executor": "@nrwl/cypress:cypress",
+    "e2e": {
+      "executor": "nx:run-commands",
+      "outputs": [],
       "options": {
-        "cypressConfig": "dep-graph/client-e2e/cypress-release.json",
-        "tsConfig": "dep-graph/client-e2e/tsconfig.e2e.json",
-        "devServerTarget": "dep-graph-client:serve-for-e2e:release",
-        "baseUrl": "http://localhost:4200"
+        "commands": [
+          "npx nx e2e-base e2e-dep-graph-client -c dev",
```

Comment:
The `dev` mode seems to be obsolete for E2E since the tests are expected to behave the same as `release` and our end-users never see it anyway. Unless I'm mistaken, I would remove that one.

Reply:
I'll try to address both of these comments with a description of what each of the test configurations does. The behavior of the client is driven by an environment file that is switched for each of these configurations. The environment file enables and disables some behavior via feature flags.

`dev` is where the majority of the tests are at this point. When served, this is the mode that contains the debug panel for aiding development and enables all experimental features. 

`release` is a small set of smoke tests that ensure that no in-development features have been left active. It checks to make sure the initial graph loads and that the debug panel and experimental features are turned off.

`watch` enables the watch mode and mocks an updated graph response when the client requests it. This checks to make sure that the client responds correctly to new projects as they're updated in the the graph.

Each of them serves a different purpose. For local development, I run the `dev` config when writing tests, and then run `watch` and `release` before raising a PR. Maybe we don't need to run all three configs as part of CI? Perhaps run all three as part of PR checks, but there's no need for a nightly run? Any opinion after seeing this context?

Reply:
That makes now sense. Thank you! I will merge this in then and we can see if some of those checks should only run on nightly.

Nightly runs across different OSs and package managers so it's vital we that on.